From dd737c006009d1b17ebb7fb4a89d2971b4d677f6 Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Thu, 19 Jun 2025 16:46:45 +0530
Subject: [PATCH 5/6] TDA54: arm64: dts: ti: k3-tda54-vdk: Enable IPC with R52+
 rproc

Add reserved memory carveouts, configure TX/RX mailboxes and enable the
DT node at board level to support IPC between A720 and R52+ rproc.

Signed-off-by: Beleswar Padhi <b-padhi@ti.com>
---
 arch/arm64/boot/dts/ti/k3-tda54-vdk.dts | 38 +++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-tda54-vdk.dts b/arch/arm64/boot/dts/ti/k3-tda54-vdk.dts
index 8c664641612f..696547d117e1 100644
--- a/arch/arm64/boot/dts/ti/k3-tda54-vdk.dts
+++ b/arch/arm64/boot/dts/ti/k3-tda54-vdk.dts
@@ -56,6 +56,18 @@ c76_memory_region: c76-memory@ac1000000 {
 			no-map;
 		};
 
+		mcu_r52ss0_core0_dma_memory_region: r52-memory@a6000000 {
+			compatible = "shared-dma-pool";
+			reg = <0x00 0xa6000000 0x00 0x100000>;
+			no-map;
+		};
+
+		mcu_r52ss0_core0_memory_region: r52-memory@a6100000 {
+			compatible = "shared-dma-pool";
+			reg = <0x00 0xa6100000 0x00 0xf00000>;
+			no-map;
+		};
+
 		secure_ddr: optee@9e800000 {
 			reg = <0x00 0x9e800000 0x00 0x01800000>;
 			no-map;
@@ -118,6 +130,16 @@ mbox_c76: mbox-c76 {
 	};
 };
 
+&mailbox0_cluster1 {
+	status = "okay";
+	interrupts = <377>;
+
+	mbox_mcu_r52ss0_core0: mbox-mcu-r52ss0-core0 {
+		ti,mbox-rx = <15 0 2>;
+		ti,mbox-tx = <3 0 2>;
+	};
+};
+
 &mcu_m55 {
 	mboxes = <&mailbox0_cluster0 &mbox_mcu_m55>;
 	memory-region = <&mcu_m55_dma_memory_region>,
@@ -131,3 +153,19 @@ &c76 {
 			<&c76_memory_region>;
 	status = "okay";
 };
+
+&mcu_r52ss0_core0 {
+	mboxes = <&mailbox0_cluster1 &mbox_mcu_r52ss0_core0>;
+	memory-region = <&mcu_r52ss0_core0_dma_memory_region>,
+			<&mcu_r52ss0_core0_memory_region>;
+	status = "okay";
+};
+
+/* LockStep mode */
+&mcu_r52ss0_core1 {
+	status = "okay";
+};
+
+&mcu_r52ss0 {
+	status = "okay";
+};
-- 
2.34.1

